mqtt/mqtt-broker/src/connection.rs

```diff
@@ -239,6 +242,12 @@ async fn incoming_task<S>(
 where
     S: Stream<Item = Result<Packet, DecodeError>> + Unpin,
 {
+    // We limit the number of incoming publications (PublishFrom) per client
+    // in order to avoid (a single) publisher to occupy whole BrokerHandle queue.
+    // This helps with QoS 0 messages throughput, due to the fact that outgoing_task
+    // also uses sends PubAck0 for QoS 0 messages to BrokerHandle queue.
+    let inflight_guard = Arc::new(Semaphore::new(10));
```

Comment:
Changes look good. But is the 10 number heuristics based? What is the thinking around the right number here?

Reply:
As discussed offline, this number was optimal in local testing.